src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -753,6 +754,9 @@ public SecurityComponents createSecurityComponents() {
      */
     @Override
     protected UserDetails authenticate2(String username, String password) throws AuthenticationException {
+        if(FIPS140.useCompliantAlgorithms() && StringUtils.isNotBlank(password) && password.length() < 14) {
```

Comment:
I believe a blank password is also not allowed.

```suggestion
        if(FIPS140.useCompliantAlgorithms() && (StringUtils.isBlank(password) || password.length() < 14)) {
```